src/server/object_services/map_builder.js

```diff
@@ -35,13 +36,14 @@ const builder_lock = new KeysLock();
 class MapBuilder {
 
     /**
-     * @param {nb.ID[]} chunk_ids 
+     * @param {nb.ID[]} chunk_ids
      * @param {nb.Tier} [move_to_tier]
+     * @param {boolean} [evict]
      */
-    constructor(chunk_ids, move_to_tier) {
+    constructor(chunk_ids, move_to_tier, evict) {
         this.chunk_ids = chunk_ids;
         this.move_to_tier = move_to_tier;
-
+        this.evict = evict;
```

Comment:
I think perhaps it would be wise to assert here that evict and move_to_tier are mutually exclusive, so that we can reduce our checks later to be simply `if (this.evict)...` instead of repeating `if (this.evict && !this.move_to_tier) ...`
```javascript
if (evict) assert.strictEqual(move_to_tier, undefined);
```